lib/Network/OAuth2/Server/API.hs

```diff
@@ -566,12 +566,28 @@ serverPostToken
     -> m Html
 -- | Revoke a given token
 serverPostToken ref user_id _ (DeleteRequest token_id) = do
-    success <- liftIO $ storeRevokeToken ref user_id token_id
-    if success then
-        let link = safeLink (Proxy :: Proxy AnchorOAuth2API) (Proxy :: Proxy ListTokens) page1
-        in throwError err302{errHeaders = [(hLocation, B.pack $ show link)]} --Redirect to tokens page
-    else
-        throwError err500{errBody = "Something quite horrible has happened"}
+    maybe_tok <- liftIO $ storeReadToken ref (Right token_id)
+    tok <- case maybe_tok of
+        Nothing -> throwError err403 { errBody = "Invalid token_id" }
```

Comment:
Perhaps a 400, you don't want the user to be able to tell between a valid and invalid token_id, though.